web/src/components/ChatConversation.tsx

```diff
@@ -41,6 +51,28 @@ const ChatConversation = ({ messages, hasError, className }: ChatConversationPro
     }
   }, [messages, messagesCount])
 
+  const typingIndicatorTimer = () => {
+    setTypingIndicatorVisible(true)
+
+    const typingIndicatorTimeout = setTimeout(() => {
+      setTypingIndicatorVisible(false)
+    }, TYPING_INDICATOR_TIMEOUT)
+
+    return () => clearTimeout(typingIndicatorTimeout)
+  }
+
+  const handleWaitingForAnswer = useCallback((answer: boolean) => {
+    if (answer) {
+      return typingIndicatorTimer()
+    }
+    return () => undefined
+  }, [])
+
+  useEffect(() => {
+    const typingIndicator = handleWaitingForAnswer(waitingForAnswer)
+    return typingIndicator
+  }, [waitingForAnswer, handleWaitingForAnswer])
```

Comment:
As discussed: please revert :) We'll just ignore the error for now :)

Reply:
Well at least it is not 11 :smile: `ChatConversation has a cyclomatic complexity of 10, threshold = 10.`